### PR TITLE
chore: add otel-builder configuration

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

- add configuration required to build a custom otel-collector with our custom receiver

**Related issue(s)**

### CODE REVIEW

**Review by rakesh-garimella — DISMISSED** [review]

File: otel-collector/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# OTEL Collector Docker Image
+
+The container image is built using the [otel builder binary](https://github.com/open-telemetry/opentelemetry-collector/tree/main/cmd/builder) based on [builder-config.yaml](https://github.com/open-telemetry/opentelemetry-collector/blob/main/cmd/otelcorecol/builder-config.yaml).
```
Comment (NHingerl) [inline otel-collector/README.md:3]:
```suggestion
The container image is built using the [OTel builder binary](https://github.com/open-telemetry/opentelemetry-collector/tree/main/cmd/builder) based on [builder-config.yaml](https://github.com/open-telemetry/opentelemetry-collector/blob/main/cmd/otelcorecol/builder-config.yaml).
```

File: otel-collector/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+This custom image has a minimal set of required receivers, processors and exporters for Kyma.
```
Comment (NHingerl) [inline otel-collector/README.md:5]:
```suggestion
This custom image has a minimal set of required receivers, processors, and exporters for Kyma.
```

File: otel-collector/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+In addition to OSS components it will contain custom receivers, processors etc contained in the [OCC repository](https://github.com/kyma-project/opentelemetry-collector-components)
```
Comment (NHingerl) [inline otel-collector/README.md:7]:
```suggestion
In addition to OSS components, it contains custom receivers, processors etc., contained in the [OCC repository](https://github.com/kyma-project/opentelemetry-collector-components).
```

File: otel-collector/README.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+The build is configured for two different scenarios:
```
[the remark this answers is not in the document]
Comment (k15r) [reply otel-collector/README.md:9]: correct, the variable is called BUILD_MODE

File: otel-collector/README.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+```
+docker build -f otel-collector/Dockerfile --build-arg GOLANG_VERSION=XXX --build-arg='OTEL_VERSION=XXX' --build-arg OTEL_CONTRIB_VERSION=XXX .
+```
+
+release-mode:
+
+```
+docker build -f otel-collector/Dockerfile --build-arg BUILD_MODE=release --build-arg GOLANG_VERSION=XXX --build-arg OTEL_VERSION=XXX --build-arg OTEL_CONTRIB_VERSION=XXX .
+```
+
```
[the remark this answers is not in the document]
Comment (k15r) [reply otel-collector/README.md:38]: well, docker will tell you

[the remark this answers is not in the document]
Comment (NHingerl) [reply otel-collector/README.md:32]: Then either let's add another step "To check whether the build was successful, see..."
or a [result](https://raw.githubusercontent.com/kyma-project/community/main/templates/resources/tutorial.md), like "In abc, you see that xyz has been created." or something like that

File: otel-collector/README.md [hunk 6]
```diff
@@ -14,19 +14,20 @@ This mode depends on the local version the OTel Collector components and can be
...
-   docker build -f otel-collector/Dockerfile --build-arg GOLANG_VERSION={XXX} --build-arg='OTEL_VERSION={XXX}' --build-arg OTEL_CONTRIB_VERSION={XXX} .
+       docker build -f otel-collector/Dockerfile --build-arg GOLANG_VERSION={XXX} --build-arg='OTEL_VERSION={XXX}' --build-arg OTEL_CONTRIB_VERSION={XXX} .
+
+   For release mode, use:
 
-release-mode:
+       docker build -f otel-collector/Dockerfile --build-arg BUILD_MODE=release --build-arg GOLANG_VERSION={XXX} --build-arg OTEL_VERSION={XXX} --build-arg OTEL_CONTRIB_VERSION={XXX} .
 
-   docker build -f otel-collector/Dockerfile --build-arg BUILD_MODE=release --build-arg GOLANG_VERSION={XXX} --build-arg OTEL_VERSION={XXX} --build-arg OTEL_CONTRIB_VERSION={XXX} .
+3. If your build was successful, the docker command will update its status output to:
```
Comment (NHingerl) [inline otel-collector/README.md:31]:
```suggestion
3. If your build was successful, the Docker command updates its status output to:
```

File: otel-collector/README.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
+# OTEL Collector Docker Image
+
+The container image is built using the [OTel builder binary](https://github.com/open-telemetry/opentelemetry-collector/tree/main/cmd/builder) based on [builder-config.yaml](https://github.com/open-telemetry/opentelemetry-collector/blob/main/cmd/otelcorecol/builder-config.yaml).
+
+This custom image has a minimal set of required receivers, processors, and exporters for Kyma.
+
+In addition to OSS components, it contains custom receivers, processors etc., contained in the [OCC repository](https://github.com/kyma-project/opentelemetry-collector-components).
+
+The build is configured for two different mode:
```
Comment (NHingerl) [inline otel-collector/README.md:9]:
```suggestion
The build is configured for two different modes:
```
sorry I didn't notice before

**Review by NHingerl — APPROVED** [review]

**Review by rakesh-garimella — APPROVED** [review]

**Pull request merged**